python/lambdas/copy-files-from-tdr/copy_files_lambda_function_test.py

```diff
@@ -246,44 +214,46 @@ def selective_error_object(bucket, key) :
+            str(ex.exception))
 
-    @patch('lambda_function.s3_client.get_object')
-    def test_should_raise_an_exception_when_UUID_is_invalid(self, mock_get_object):
-        mock_get_object.return_value = {
+    def test_should_raise_an_exception_when_UUID_is_invalid(self):
+        mock_get_object_response = {
             'Body':
                 '{"ConsignmentReference": "TDR-2024-PQXN", "FileReference": "ZDSCFC", "Series":"MOCK1 123", '
                 '"TransferInitiatedDatetime": "2024-09-19 07:21:57", "UUID": "invalid-uuid-format"}'
-             }
-        with self.assertRaises(Exception) as ex:
-            lambda_function.validate_formats('any_bucket_patched', 'any_key_patched')
-        self.assertEqual("Unable to parse UUID, 'invalid-uuid-format'. Invalid format" , str(ex.exception))
-
-
-    @patch('lambda_function.s3_client.get_object')
-    def test_should_raise_an_exception_when_transfer_initiated_date_is_invalid(self, mock_get_object):
-        mock_get_object.return_value = {
-            'Body':
-                '{"ConsignmentReference": "TDR-2024-PQXN", "FileReference": "ZDSCFC", "Series":"MOCK1 123", '
-                '"TransferInitiatedDatetime": "2024-19-19 07:21:57", "UUID": "bb7bb923-b82c-4203-a3c1-ea3f362ef4da"}'
-             }
+        }
         with self.assertRaises(Exception) as ex:
-            lambda_function.validate_formats('any_bucket_patched', 'any_key_patched')
-        self.assertEqual("Unable to parse date, '2024-19-19 07:21:57'. Invalid format" , str(ex.exception))
+            lambda_function.validate_formats(mock_get_object_response, 'any_bucket_patched', 'any_key_patched')
+        self.assertEqual("Unable to parse UUID, 'invalid-uuid-format' from file 'any_key_patched' in "
+                         "bucket 'any_bucket_patched'. Invalid format", str(ex.exception))
 
-    @patch('lambda_function.s3_client.get_object')
-    def test_should_raise_an_exception_when_series_is_empty(self, mock_get_object):
-        mock_get_object.return_value = {
+    def test_should_raise_an_exception_when_series_is_empty(self):
+        mock_get_object_response = {
             'Body':
                 '{"ConsignmentReference": "TDR-2024-PQXN", "FileReference": "ZDSCFC", "Series":" ", '
                 '"TransferInitiatedDatetime": "2024-09-19 07:21:57", "UUID": "bb7bb923-b82c-4203-a3c1-ea3f362ef4da"}'
-             }
+        }
         with self.assertRaises(Exception) as ex:
-            lambda_function.validate_formats('any_bucket_patched', 'any_key_patched')
-        self.assertEqual("Empty series value, unable to proceed" , str(ex.exception))
+            lambda_function.validate_formats(mock_get_object_response, 'any_bucket_patched', 'any_key_patched')
+        self.assertEqual(
+            "Empty Series value in file 'any_key_patched' in bucket 'any_bucket_patched'. Unable to proceed",
+            str(ex.exception))
+
+    # def test_should_raise_an_exception_when_series_format_is_invalid(self):
```

Comment:
Did you mean to comment this out?

Reply:
yes, it is related to FIXME in the implementation